overlord/devicestate/firstboot.go

```diff
@@ -128,18 +129,25 @@ func populateStateFromSeedImpl(st *state.State) ([]*state.TaskSet, error) {
 	tsAll := []*state.TaskSet{}
 	configTss := []*state.TaskSet{}
 
-	// if there are snaps to seed, core needs to be seeded too
+	baseSnap := "core"
+	if model.Base() != "" {
+		baseSnap = model.Base()
+	}
+	// if there are snaps to seed, core/base needs to be seeded too
 	if len(seed.Snaps) != 0 {
```

Comment:
to be consistent if base is not "core" we need code to make sure "snapd" is installed first here, so   snapd, then base etc